lxd/instances.go

```diff
@@ -486,9 +530,48 @@ func instancesShutdown(instances []instance.Instance) {
 		maxConcurrent = instCount
 	}
 
+	// Start the busy instance tracker if instancesToOps is provided.
+	busyInstancesCh := make(chan instance.Instance)
+	if len(instancesToOps) > 0 {
+		go func() {
+			// Map to track busy instances
+			busyInstances := make(map[string]instance.Instance)
+			ticker := time.NewTicker(time.Second)
+			defer ticker.Stop()
+
+			for {
+				select {
+				case inst, ok := <-busyInstancesCh:
+					if !ok {
+						logger.Debug("Finishing busy instances tracking")
+						return
+					}
+
+					logger.Debug("Instance received for busy tracking", logger.Ctx{"instance": inst.Name()})
+					instanceURL := entity.InstanceURL(inst.Project().Name, inst.Name()).String()
+					busyInstances[instanceURL] = inst
+				case <-ticker.C:
+					ctxErr := ctx.Err()
+					if ctxErr != nil {
+						logger.Info("Skip waiting for busy instances to complete")
+					}
+
+					for instanceURL, inst := range busyInstances {
+						if ctxErr != nil || !isInstanceBusy(inst, instancesToOps, &instancesToOpsMu) {
+							delete(busyInstances, instanceURL)
+							logger.Debug("Instance removed from busy tracking, sending to shutdown channel", logger.Ctx{"instance": inst.Name()})
+							instShutdownCh <- inst
+						}
+					}
+				}
+			}
+		}()
+	}
+
 	for i := 0; i < maxConcurrent; i++ {
 		go func(instShutdownCh <-chan instance.Instance) {
 			for inst := range instShutdownCh {
+				logger.Debug("Instance received for shutdown", logger.Ctx{"project": inst.Project().Name, "instance": inst.Name(), "timestamp": time.Now().UnixNano()})
```

Comment:
@gabrielmougard are the new `timestamp` nano second fields required, they are rather noisy.